beluga_tutorial/src/main.cpp

```diff
@@ -0,0 +1,258 @@
+    tutorial_params.velocity = params["velocity"].as<double>();
+    tutorial_params.translation_sigma = params["translation_sigma"].as<double>();
+    tutorial_params.sensor_range = params["sensor_range"].as<double>();
+    tutorial_params.sensor_model_sigam = params["sensor_model_sigam"].as<double>();
+    tutorial_params.min_particle_weight = params["min_particle_weight"].as<double>();
+  } catch (YAML::BadFile& e) {
+    std::cout << e.what() << "\n";
+  }
+}
+
+void load_landmark_map_from_yaml(LandmarkMapVector& landamrk_map) {
+  try {
+    YAML::Node node = YAML::LoadFile("./src/beluga/beluga_tutorial/params/landmark_map.yaml");
+    landamrk_map = node["landamrk_map"].as<LandmarkMapVector>();
+  } catch (YAML::BadFile& e) {
+    std::cout << e.what() << "\n";
+  }
+}
+
+void save_tutorial_dataset_to_yaml(const std::string& file_name, const TutorialDataset& tutorial_dataset) {
+  YAML::Node node;
+
+  for (const auto& lmp : tutorial_dataset.landmark_map) {
+    node["landamrk_map"].push_back(lmp);
+  }
+
+  for (auto&& [cycle, data] : tutorial_dataset.sim_data | ranges::views::enumerate) {
+    node["simulation_dataset"][cycle]["ground_truth"] = data.ground_truth;
+
+    for (const auto& current : data.current) {
+      node["simulation_dataset"][cycle]["particles"]["current"]["states"].push_back(std::get<0>(current));
+      node["simulation_dataset"][cycle]["particles"]["current"]["weights"].push_back(
+          static_cast<double>(std::get<1>(current)));
+    }
+
+    for (const auto& propagate : data.propagate) {
+      node["simulation_dataset"][cycle]["particles"]["propagate"]["states"].push_back(std::get<0>(propagate));
+      node["simulation_dataset"][cycle]["particles"]["propagate"]["weights"].push_back(
+          static_cast<double>(std::get<1>(propagate)));
+    }
+
+    for (const auto& reweight : data.reweight) {
+      node["simulation_dataset"][cycle]["particles"]["reweight"]["states"].push_back(std::get<0>(reweight));
+      node["simulation_dataset"][cycle]["particles"]["reweight"]["weights"].push_back(
+          static_cast<double>(std::get<1>(reweight)));
+    }
+
+    for (const auto& resample : data.resample) {
+      node["simulation_dataset"][cycle]["particles"]["resample"]["states"].push_back(std::get<0>(resample));
+      node["simulation_dataset"][cycle]["particles"]["resample"]["weights"].push_back(
+          static_cast<double>(std::get<1>(resample)));
+    }
+
+    node["simulation_dataset"][cycle]["estimation"]["mean"] = std::get<0>(data.estimation);
+    node["simulation_dataset"][cycle]["estimation"]["sd"] = std::get<1>(data.estimation);
+  }
+
+  const std::string bags_path{"./src/beluga/beluga_tutorial/bags/"};
+  std::filesystem::create_directory(bags_path);
+  std::ofstream fout(bags_path + file_name, std::ios::trunc);
+  fout << node;
+  fout.close();
+}
+
+int main() {
+  // Load params from yaml file
+  TutorialParams tutorial_params;
+  load_params_from_yaml(tutorial_params);
```

Comment:
@Alondruck meta: I'm not super fond of hard-coded paths. I don't want the tutorial to get any bigger either. Would it make sense to put all paths in the configuration file and take the path to the configuration file as the only command-line argument? If none is provided, we can use `TutorialParams` default constructor to populate it with sensible defaults.

Reply:
Yes, I agree with you. I forgot to mark this as a TODO earlier. So just to clarify, the command line argument will supply me with the path to the config file, and for all other paths in the code, I'll add them as parameters. Is that correct?

Reply:
Right. No argument parsing here, just check if `argc > 1` and get the path from `argv[1]`.